src/schemabuilder.rs

```diff
@@ -203,7 +204,27 @@ impl SchemaBuilder {
     ) -> PyResult<Self> {
         let builder = &mut self.builder;
 
-        let opts = SchemaBuilder::build_int_option(stored, indexed, fast)?;
+        let mut opts = DateOptions::default();
+        if stored {
+            opts = opts.set_stored();
+        }
+        if indexed {
+            opts = opts.set_indexed();
+        }
+        let fast = match fast {
+            Some(f) => {
+                let f = f.to_lowercase();
+                match f.as_ref() {
+                    "single" => Some(schema::Cardinality::SingleValue),
+                    "multi" => Some(schema::Cardinality::MultiValues),
+                    _ => return Err(exceptions::PyValueError::new_err(
+                        "Invalid index option, valid choices are: 'multivalue' and 'singlevalue'"
+                    )),
+                }
+            }
+            None => None,
+        };
+        opts = opts.set_fast(fast.unwrap());
```

Comment:
```suggestion
        let cardinality = match fast {
            Some(f) => {
                let f = f.to_lowercase();
                match f.as_ref() {
                    "single" => Some(schema::Cardinality::SingleValue),
                    "multi" => Some(schema::Cardinality::MultiValues),
                    _ => return Err(exceptions::PyValueError::new_err(
                        "Invalid index option, valid choices are: 'multivalue' and 'singlevalue'"
                    )),
                }
            }
            None => None,
        };
        if fast.is_some(){
            opts = opts.set_fast(fast.unwrap());
        }
```

I tried these changes out with some sample projects and this seems to fail in case fast is 'None', as 
`pyo3_runtime.PanicException: called `Option::unwrap()` on a `None` value`.

Apart from this looks good!

Reply:
Also you can format rust code using
`rustfmt --check src/*.rs`.

Reply:
@Sidhant29  Good finds, I updated to put in the fix you suggested, and applied rustfmt.